### DISCUSSION TITLE
Code review doesn't allow commenting on commit messages

### DISCUSSION THREAD

**Discussion by InterLinked1** [community/community#56449] [Pull Requests]
### Select Topic Area

Bug

### Body

A large open source project to which I contribute recently migrated from Gerrit to GitHub.

There are several things that are significantly more challenging or inconvenient now as a result, but I just encountered one that seems insurmountable.

In Gerrit, we can easily comment on the commit message just as with any other changed file.

GitHub does not seem to have any way of doing this. I would like to comment on specific lines on the commit message, and am unable to do so.

This is a major bug and deficiency in the code review functionality. There is no elegant workaround for this. Reviewers MUST be able to comment on commit messages as with any other code. This is essential for requesting changes to the commit message, with an inline comment in the relevant place.

**Comment by rjarry** [comment]
Hi, I wouldn't categorize this as a bug per se but it is a crucial feature that is currently lacking.

This is probably one of the reason why [some contributors on github don't care about their git commit messages](https://blog.trunk.io/git-commit-messages-are-useless-c2f3c46f678e). They stuff everything into the pull request description and leave the git history aside.

**Reply by MasterInQuestion** [reply]
https://trunk.io/blog/git-commit-messages-are-useless
    The primary issue is the inability to edit commit history: once cast irreversible.
    Which tends to impose an unnecessary burden on the authors.

**Comment by coiby** [comment]
Yes, allowing commenting on commit messages is what my team needs we value the commit message in our workflow.

**Comment by coiby** [comment]
In case it interests anyone, here is [the Gitlab counterpart](https://gitlab.com/gitlab-org/gitlab/-/issues/19691).

**Comment by ADKaster** [comment]
Related to an older discussion Q: https://github.com/orgs/community/discussions/4983

**Comment by MasterInQuestion** [comment]
"elegant workaround for this"
    Several examples of my practice:
    [ https://github.com/MasterInQuestion/coordTransform/commit/bab1643d45a60254b47c503c3cce97b167d96128](https://github.com/MasterInQuestion/coordTransform/commit/bab1643d45a60254b47c503c3cce97b167d96128)
    [ https://github.com/MasterInQuestion/ShellUtils/commit/34910eb5ca7ff55f02d337f46aa75c031277ec66](https://github.com/MasterInQuestion/ShellUtils/commit/34910eb5ca7ff55f02d337f46aa75c031277ec66)

    See also: https://github.com/orgs/community/discussions/126301#discussioncomment-10436744

**Comment by arut** [comment]
Completely agree. We in nginx are really missing this feature after migration to GitHub. We care a lot about commit logs.

**Comment by serhii-surazhskyi-gl** [comment]
Completely agree and even created a bug report... lets see if that can be changed.

Edited: and the bug report was archived... ha-ha.
